File: javascript/bubbleSort.js

function bubbleSort(array) {
  let counter = 0
  let sorted = false
  while (!sorted) {
    sorted = true
    for (let i = 0; i < array.length - 1 - counter; i++) {
      if (array[i] > array[i + 1]) {
        swap(array, i, i + 1)
        sorted =  false
      }
    }
  }
  return array
}

function swap(array, i, j) {
  let tmp = array[j]
  array[j] = array[i]
  array[i] = tmp
}

exports.bubbleSort = bubbleSort

File: python/bubbleSort.py

def bubbleSort(array):
  isSorted = False
  counter = 0
  while not isSorted:
    isSorted = True
    for i in range(len(array) - 1 - counter ):
      if array[i] > array[i + 1]:
        swap(array, i, i + 1)
        isSorted = False
  return array

def swap(array, i, j):
  tmp = array[i]
  array[i] = array[j]
  array[j] = tmp
